### PR TITLE
Add TypeScript docs for APIs

### PR DESCRIPTION
**Pull request by SanjulaGanepola**

### CODE REVIEW

File: src/iproject.ts [hunk 1]
```diff
@@ -44,42 +104,87 @@ export class IProject {
...
     return Uri.file(path.join(directory ? directory.fsPath : this.workspaceFolder.uri.fsPath, logDirectory, type));
   }
 
+  /**
+   * Check if a project file exists. *Note* that the `directory` parameter should
+   * be passed in when the desired project file is a directory level `.ibmi.json`
+   * that is not in the root directory.
+   * 
+   * @param type A type of project file.
+   * @param directory The directory containing the project file.
```
Comment (edmundreinhardt) [inline src/iproject.ts:138]: is this a relative to project root directory or absolute path?

File: src/iproject.ts [hunk 2]
```diff
@@ -17,22 +17,82 @@ import { IBMiObject } from "@halcyontech/vscode-ibmi-types";
...
+  /**
+   * An object that represents the contents of the project's `iproj.json` with
+   * variables resolved based on the project's `.env` file.
+   */
   private state: IProjectT | undefined;
+
+  /**
+   * A map of directory locations and the content of each `.ibmi.json` file in
+   * a project. *Note* that for the root `.ibmi.json` file, the contents of the
+   * `iproj.json` file will take precedence.
```
Comment (edmundreinhardt) [inline src/iproject.ts:73]: Let us know if the Type has variables resolved or not

**Review by edmundreinhardt — CHANGES_REQUESTED** [review]

File: src/iproject.ts (hunk 2 above)
Comment (SanjulaGanepola) [reply src/iproject.ts]: The variables are not resolved in the buildMap. I have updated the TS doc

File: src/iproject.ts (hunk 1 above)
Comment (SanjulaGanepola) [reply src/iproject.ts:139]: Since we are dealing with a `uri` here, it will always be absolute.

**Review by edmundreinhardt — APPROVED** [review]
thank you

**Pull request merged**